### ISSUE TITLE
PGP/MIME

### ISSUE THREAD

**Issue by alberto743** [mailvelope/mailvelope#41]
Is there a plan to support PGP/MIME (sign&encrypt), including with attachments?

**Comment by rubin110** [comment]
Thank you for adding some support for incoming PGP/MIME messages. Any chance there'll be love for out going messages too? This is now the only thing stopping me from dropping Thunderbird.

**Comment by mdik** [comment]
@rubin110 what do you mean by "Thank you for adding some support for incoming PGP/MIME messages."? I don't see it (,yet?), but I also would be very happy if Mailvelope would follow the attachment links to be able to decrypt PGP/MIME messages!

**Comment by HibikiTaisuna** [comment]
As there is no way to read an PGP/MIME attachment I propose adding a file encryption/decryption support to Mailvelope. That way we can still decide to make our own attachements for sending and receive PGP/MIME attachments without relying on a different keystore.

**Comment by mdik** [comment]
How is there no way to read a PGP/MIME attachment? Aren't they just links you can pull the contents from?

**Comment by HibikiTaisuna** [comment]
I am referring to the current support in Mailvelope.

**Comment by lispeaking** [comment]
For gmail there is this 'show original' item in a pull down menu. The full message source might be a good starting point to decrypt PGP-MIME.

**Comment by Sigmun** [comment]
One up for reading or opening PGP/MIME.

**Comment by dashohoxha** [comment]
:+1:  I agree that PGP/MIME functionality would be great.

Are you already using this: https://www.inboxsdk.com/ ?
Or maybe it doesn't help, or is totally irrelevant (excuse me for my ignorance).

**Comment by paulmering** [comment]
My e-mail provider mail.de offers automatic encryption of incoming (not end-to-end encrypted) mails - but I cannot open them in the webmailer because of missing PGP/MIME support in mailvelope - so this feature would be great for me as well :-)

**Comment by toberndo** [comment]
@paulmering Mailvelope supports already reading PGP/MIME if the PGP armored block is displayed in the mail body (or a preview window).

**Comment by Kristof11** [comment]
:+1:  Would love to see a quiet option so I can sign my emails always instead of only to people from who I know that they're using PGP.
For non-PGP users those emails seems odd.

**Comment by woj-tek** [comment]
@toberndo could you expound of that? I've tried that with opening signature attachment withing gmail but nothing happened.

If would be great if it could work!

**Comment by j-rahman** [comment]
PGP/MIME signatures come as attachments, and Mailvelope does not recognise them.

**Comment by Sowiedu** [comment]
Still an issue, I love mailvelope other than that. Can you please do something about it? :(

**Comment by junwei-wang** [comment]
One possible idea is to let the users tick an option to read their Gmail API.
If they would like to access to read / write attachments, they have to let their data exposed to the extension. Besides, the privacy policy should be well-written.

**Comment by xnox** [comment]
```
$ cat Test\ encrypted\ mail.eml | gpg --verify
gpg: malformed CRC
gpg: encrypted_mdc packet with unknown version 255
gpg: no signature found
gpg: quoted printable character in armour - probably a buggy MTA has been used
gpg: the signature could not be verified.
```

Looks like there is garbage generated in the email. Not sure by who though

**Comment by fdosani** [comment]
Curious about the feature too. Having issues with startmail not being able to read/detect encrypted emails from mailvelope (gmail).
Seems like PGP/Inline vs PGP/MIME (if I'm not mistaken) currently causes it to not work.

**Comment by ioogithub** [comment]
I believe I have this problem too. I am an new Nextcloud user and I have been trying to get Nextcloud mail and Mailvelope working without success all day. Mailvelope works perfectly with my mail provider's webmail page but doesn't work at all with Nextcloud. All I see is a single attachment "encrypted.asc".

Many mail providers will not switch to inline because as vasyugan mentioned in this thread: https://github.com/nextcloud/mail/issues/3833,  PGP/Mime is the standard best practice while inline is a depreciated and not well supported.

If many webmail providers have been seamlessly doing this for years then the problem is not unknown, it is just not supported currently in Nextcloud.

In the linked thread, ChristophWurst mentioned that:
> So it's (development) is blocked until we found the time to work on this or a customer funds the development.

I am not an enterprise customer however I may be interested in funding this development. Who do I speak to about this? It is important because without it, Nextcloud mail does not work at all for me.

**Comment by pravi** [comment]
Has this been considered? At least for the actual text, even if not for attachments https://josefsson.org/inline-openpgp-considered-harmful.html - What should I be doing? section at the end gives an example, would mailvelope be able to use that format?

Can mailvelope modify mail headers? Or can it only inject text within body? At the least, it can remove the signature from html part to make Thunderbird happy.

**Comment by toberndo** [comment]
We have support for PGP/MIME if the mail provider implements Mailvelope's client-API. Or if you use the Gmail API in Mailvelope. For other scenarios, where Mailvelope has only access to the content of the email via the webmail UI you cannot read the MIME structure of the email, nor can you create it when sending an email.

**Comment by pravi** [comment]
> We have support for PGP/MIME if the mail provider implements Mailvelope's client-API. Or if you use the Gmail API in Mailvelope. For other scenarios, where Mailvelope has only access to the content of the email via the webmail UI you cannot read the MIME structure of the email, nor can you create it when sending an email.

It was my friend who had to authorize mailvelope in gmail, so I assume it used gmail api. But the mail was sent with content type text/plain and it came as inline PGP.

When mail sent with thunderbird it has proper pgp/mime headers, for example, mail sent with mailvelope has
`Content-Type: text/plain; charset="UTF-8"`

whereas mail sent from thunderbird has,
```
Content-Type: multipart/signed; micalg=pgp-sha256;
 protocol="application/pgp-signature";
 boundary="------------RXNCLUafAyfsb5fm7Ia92WoS"
```

**Comment by toberndo** [comment]
Right, that's a limitation the Gmail-API, that does not support setting the content type of the first MIME node to multipart/encrypted (that's what we tested) and I guess it should be the same for multipart/signed.

**Comment by wiktor-k** [comment]
> Right, that's a limitation the Gmail-API, that does not support setting the content type of the first MIME node to multipart/encrypted (that's what we tested) and I guess it should be the same for multipart/signed.

Just for completeness this Gmail issue [has been reported at least 5 years ago](https://github.com/FlowCrypt/flowcrypt-browser/issues/3157#issuecomment-735755448) and they still haven't fixed it :-(
